### PR TITLE
feat: edit activity + tests

### LINKED ISSUE #185: Implement edit activity when pressing on an activity card
When clicking on an activity card, in Activities screen or ActivitiesForOneDay screen, the Edit Activity screen should open and users should be able to edit the activity. The Edit Activity Screen is similar to the Add Activity screen

### PR DESCRIPTION
**Pull request by mirceagavrizi**
## Summary

This PR introduces the feature of editing an existing activity. For each activity item, I added an edit button near the delete button which leads the user to editing the activity. The edit activity screen calls the add activity screen with an extra parameter, the already existing activity and makes the required changes.

I will attach a screen recording which presents the behavior.

## Changes

- **Models:** Added tripsViewModel and navigationActions as parameters to all the activity related screens/methods
- **Screens/UI:** The edit activity screen calls the add activity screen with the fields already completed
- **Bug Fixes/Improvements:** Addresses #185 .

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [x] This PR resolves #185.
- [x] Tests have been added for new functionality.
- [x] Screenshots or UI changes have been attached (if applicable).
- [x] Documentation has been updated (if applicable).

##  Testing

Explain how this feature was tested and what kind of testing (unit, integration, UI) has been done. Mention platforms or configurations used for testing.
- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this

If you haven't added tests, please create a new issue and assign it.

##  Related Issues/Tickets

Closes #185 

## Screenshots (if applicable)

https://github.com/user-attachments/assets/c29f2c67-731a-4c49-ad5c-602fe7a66508

## 💡 Additional Context

- we need to decide for when editing a trip or an activity how we'll address the dates(i.e. if the date is now in the past the trip/activity can't be saved now)

### CODE REVIEW

File: app/src/main/java/com/android/voyageur/ui/trip/activities/ActivityItem.kt [hunk 1]
```diff
@@ -90,21 +96,38 @@ fun ActivityItem(
...
+                            //                              Toast.makeText(
+                            //                                  context,
+                            //                                  "Edit activity not implemented yet",
+                            //                                  Toast.LENGTH_SHORT)
+                            //                                  .show()
+                          }) {
+                            Icon(
+                                imageVector = Icons.TwoTone.Edit,
+                                contentDescription = "Edit Activity",
+                                tint = Color.Green)
```
Comment (ioanajianu) [inline app/src/main/java/com/android/voyageur/ui/trip/activities/ActivityItem.kt:115]: I think this color is a bit too bright for the rest of the app, and green isn't really suggestive for the Edit action. Could you leave the default color instead? (the same color as the expand icon for example)

**Review by ioanajianu — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/android/voyageur/ui/trip/activities/ActivityItem.kt (hunk 1 above)
Comment (mirceagavrizi) [reply app/src/main/java/com/android/voyageur/ui/trip/activities/ActivityItem.kt:115]: Changed it.

File: app/src/main/java/com/android/voyageur/ui/trip/activities/EditActivity.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    tripsViewModel: TripsViewModel,
+) {
+  val activity by tripsViewModel.selectedActivity.collectAsState()
+  val selectedActivity = activity!!
+
+  Scaffold(
+      modifier = Modifier.testTag("editActivityScreen"),
+      content = { pd ->
+        Box(modifier = Modifier.padding(pd)) {
+          AddActivityScreen(tripsViewModel, navigationActions, existingActivity = selectedActivity)
```
[the remark this answers is not in the document]
Comment (mirceagavrizi) [reply app/src/main/java/com/android/voyageur/ui/trip/activities/EditActivity.kt:29]: Fixed it, now the fields are empty when editing a draft.

**Review by ioanajianu — APPROVED** [review]
Thanks for implementing this and taking into account my previous comments! Looks great, thanks Mircea!

**Pull request merged**